packages/components/src/utils/headless-cms/HeadlessCMSHelper.tsx

```diff
@@ -0,0 +1,239 @@
+import format from 'date-fns/format';
+import React from 'react';
+import type {
+  ArticleType,
+  Category,
+  Categories,
+  Config as RCHCConfig,
+  PageType,
+  GeneralCollectionType,
+  CardProps,
+  LanguageCodeEnum,
+} from 'react-helsinki-headless-cms';
+import {
+  Card,
+  getCollections,
+  Collection,
+  EventSearchCollection,
+  EventSelectionCollection,
+  getCollectionUIType,
+  isEventSearchCollection,
+  isEventSelectionCollection,
+  isPageType,
+  isArticleType,
+  getArticlePageCardProps as getArticlePageCardPropsBase,
+  ModuleItemTypeEnum,
+  isLocationsSelectionCollection,
+  LocationsSelectionCollection,
+} from 'react-helsinki-headless-cms';
+import type { AppLanguage } from '../../types';
+
+export interface HeadlessCMSApp {
+  cmsArticlesContextPath: string;
+  cmsPagesContextPath: string;
+  dateFormat: string;
+  ArticleDetails: React.FC<{ keywords?: string[] }>;
+}
+
+export class HeadlessCMSHelper {
+  cmsArticlesContextPath: string;
+  cmsPagesContextPath: string;
+  dateFormat: string;
+  ArticleDetails: React.FC<{ keywords?: string[] }>;
+
+  constructor({
+    cmsArticlesContextPath,
+    cmsPagesContextPath,
+    dateFormat,
+    ArticleDetails,
+  }: HeadlessCMSApp) {
+    this.cmsArticlesContextPath = cmsArticlesContextPath;
+    this.cmsPagesContextPath = cmsPagesContextPath;
+    this.dateFormat = dateFormat;
+    this.ArticleDetails = ArticleDetails;
+  }
+
+  // NOTE: Referenced 2 times in next-pages
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  getUriID = (slugs: string[], locale: AppLanguage): string => {
```

Comment:
I would rather use class functions inside a class instad of arrow functions, unless you plan to destruct the function from the class and pass the reference to it?
`getUriID(slugs: string[], locale: AppLanguage): string { ... }`

Reply:
https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1